crates/client/block_import/src/rayon.rs

```diff
@@ -19,26 +22,44 @@ impl Default for RayonPool {
 impl RayonPool {
     pub fn new() -> Self {
         let n_cores = thread::available_parallelism().expect("Getting the number of cores").get();
-
         let max_tasks = n_cores * 2;
-        Self { semaphore: Semaphore::new(max_tasks) }
+        Self { semaphore: Semaphore::new(max_tasks), max_tasks, permit_id: 0.into(), n_acquired_permits: 0.into() }
     }
 
     pub async fn spawn_rayon_task<F, R>(&self, func: F) -> R
     where
         F: FnOnce() -> R + Send + 'static,
         R: Send + 'static,
     {
-        let _permit = self.semaphore.acquire().await;
+        let max_tasks = self.max_tasks;
+        let permit_id = self.permit_id.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        log::debug!("acquire permit {permit_id}");
+        let permit = self.semaphore.acquire().await.expect("Poisoned semaphore");
+        let n_acquired_permits = self.n_acquired_permits.fetch_add(1, std::sync::atomic::Ordering::SeqCst) + 1;
```

Comment:
Are `permit` and `n_aquired_permits` still needed? They seem to be only used for debug output.

Reply:
yep it's for debug output but i was thinking of letting them here just in case, if we ever need to debug this part of the code again?

Reply:
this should have no performance impact, this is called only once per block

Reply:
Yea fait :)